test/lib/ansible_test/_util/controller/sanity/code-smell/runtime-metadata.py

```diff
@@ -264,6 +274,22 @@ def validate_metadata_file(path, is_ansible, check_deprecation_dates=False):
     list_dict_import_redirection_schema = [{str_type: import_redirection_schema}
                                            for str_type in string_types]
 
+    # action_groups schema
+
+    def at_most_one_dict(value):
+        if isinstance(value, Sequence):
+            if sum(1 for v in value if isinstance(v, Mapping)) > 1:
+                raise Invalid('List mut contain at most one dictionary')
```

Comment:
Thanks for adding this.

```suggestion
                raise Invalid('List must contain at most one dictionary')
```